scala-package/core/src/main/scala/ml/dmlc/mxnet/module/BaseModule.scala

```diff
@@ -0,0 +1,552 @@
+   * @param batchEndCallback Each callback will be called with a `BatchEndParam`.
+   * @param kvstore Default `'local'`.
+   * @param optimizer
+   * @param evalEndCallback These will be called at the end of each full evaluation,
+   *                        with the metrics over the entire evaluation set.
+   * @param evalBatchEndCallback These will be called at the end of each minibatch during evaluation
+   * @param initializer Will be called to initialize the module parameters
+   *                    if not already initialized.
+   * @param argParams Default `None`, if not `None`, should be existing parameters from a trained
+   *                  model or loaded from a checkpoint (previously saved model). In this case,
+   *                  the value here will be used to initialize the module parameters,
+   *                  unless they are already initialized by the user
+   *                  via a call to `init_params` or `fit`.
+   *                  `arg_params` has higher priority to `initializer`.
+   * @param auxParams Default `None`. Similar to `argParams`, except for auxiliary states.
+   * @param allowMissing Default `False`. Indicate whether we allow missing parameters
+   *                     when `arg_params` and `aux_params` are not `None`.
+   *                     If this is `True`, then the missing parameters will be
+   *                     initialized via the `initializer`.
+   * @param forceRebind Default `False`. Whether to force rebinding the executors if already binded.
+   * @param forceInit Default `False`. Indicate whether we should force initialization even if the
+   *                  parameters are already initialized.
+   * @param beginEpoch Default `0`. Indicate the starting epoch. Usually, if we are resuming from a
+   *                   checkpoint saved at a previous training phase at epoch N,
+   *                   then we should specify this value as N+1.
+   * @param numEpoch Number of epochs to run training.
+   */
+  // scalastyle:off parameterNum
+  def fit(trainData: DataIter, evalData: Option[DataIter] = None,
+          evalMetric: EvalMetric = new Accuracy(),
+          epochEndCallback: Option[EpochEndCallback] = None,
+          batchEndCallback: Option[BatchEndCallback] = None,
+          kvstore: String = "local",
+          optimizer: Optimizer = new SGD(),
+          evalEndCallback: Option[BatchEndCallback] = None,
+          evalBatchEndCallback: Option[BatchEndCallback] = None,
+          initializer: Initializer = new Uniform(0.01f),
+          argParams: Map[String, NDArray] = null,
+          auxParams: Map[String, NDArray] = null,
+          allowMissing: Boolean = false,
+          forceRebind: Boolean = false,
+          forceInit: Boolean = false,
+          beginEpoch: Int = 0, numEpoch: Int = 1,
+          validationMetric: Option[EvalMetric] = None,
+          monitor: Option[Monitor] = None): Unit = {
```

Comment:
....fishy to have such long parameter list....define a parameter container class?